src/main/java/com/nutomic/syncthingandroid/fragments/SelectFolderFragment.java

```diff
@@ -0,0 +1,75 @@
+package com.nutomic.syncthingandroid.fragments;
+
+import android.database.Cursor;
+import android.net.Uri;
+import android.provider.MediaStore;
+import android.util.Log;
+import android.view.View;
+import android.widget.AdapterView;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.ArrayList;
+
+public class SelectFolderFragment extends FoldersFragment {
+
+    private static final String TAG = "SelectFolderFragment";
+
+    @Override
+    public void onItemClick(AdapterView<?> adapterView, View view, int i, long l) {
+        for (Uri uri : files) {
```

Comment:
You are currently doing this on the GUI thread, but it should be done in a background thread (probably with an AsyncTask).